src/main/java/ch/njol/skript/expressions/ExprIndicesOfX.java

```diff
@@ -0,0 +1,159 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.*;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.util.LiteralUtils;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Name("Indices of X in List")
+@Description(
+		"Returns the indices or positions of a list where the value at that index is the provided value. " +
+		"Getting the indices of a list will return strings, whereas the positions will return a number."
+)
+@Examples({
+		"set {_list::*} to 1, 2, 3, 1, 2, 3",
+		"set {_indices::*} to the indices of the value 1 in {_list::*}",
+		"# {_indices::*} is now \"1\", \"4\"",
+		"",
+		"set {_indices::*} to the indices of the value 2 in {_list::*}",
+		"# {_indices::*} is now \"2\", \"5\"",
+		"",
+		"set {_positions::*} to the positions of the value 3 in {_list::*}",
+		"# {_positions::*} is now 3, 6",
+		"",
+		"set {_otherlist::burb} to 100",
+		"set {_otherlist::burp} to 100",
+		"set {_otherlist::brup} to 100",
+		"set {_indices::*} to the first index of the value 100 in {_otherlist::*}",
+		"# {_indices::*} is now \"burb\"",
+		"set {_indices::*} to the last index of the value 100 in {_otherlist::*}",
+		"# {_indices::*} is now \"burp\"",
+		"",
+		"set {_positions::*} to the positions of the value 100 in {_otherlist::*}",
+		"# {_positions::*} is now 1, 2, 3",
+})
+@Since("INSERT VERSION")
+public class ExprIndicesOfX extends SimpleExpression<Object> {
+
+	static {
+		Skript.registerExpression(ExprIndicesOfX.class, Object.class, ExpressionType.COMBINED,
+			"[the] [1:first|2:last] (indices|index[es]|:position[s]) of [[the] value] %object% in %objects%"
+		);
+	}
+
+	private IndexType type;
+	private boolean position;
+	private Expression<?> value;
+	private Expression<?> objects;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (exprs[1].isSingle() || (exprs[1] instanceof Variable<?> var && !var.isList())) {
```

Comment:
the first check covers the second, so the second is redundant

Reply:
ooo that's cool, i see